### PR TITLE
Set `RotaryEmbedding` base frequency from config

### PR DESCRIPTION
**Pull request by shan18**
### What does this PR do?

Add the ability to change the base value of RoPE embeddings from config.

**Collection**: nlp

### CODE REVIEW

File: nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py [hunk 1]
```diff
@@ -384,6 +384,7 @@ def model_provider_func(self, pre_process, post_process):
                 use_flash_attention=self.cfg.get('use_flash_attention', False),
                 megatron_legacy=self.cfg.get('megatron_legacy', False),
                 seq_len_interpolation_factor=self.cfg.get('seq_len_interpolation_factor', None),
+                base=self.cfg.get('base', 10000),
```
Comment (hsiehjackson) [inline nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py:387]: Should we change the name to `rotary_base` (similar to `rotary_percentage`), so people can know what is this param used for?

**Review by hsiehjackson — COMMENTED** [review]
Overall, looks good to me! Leave a minor comment.

File: nemo/collections/nlp/models/language_modeling/megatron/gpt_model.py [hunk 2]
```diff
@@ -167,6 +167,7 @@ def __init__(
         ub_tp_comm_overlap=False,
         use_flash_attention=False,
         seq_len_interpolation_factor=None,
+        base=10000,
```
Comment (titu1994) [inline nemo/collections/nlp/models/language_modeling/megatron/gpt_model.py:170]: Give it better name - rotary_base or base_positional_scale or something

File: nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py (hunk 1 above)
Comment (shan18) [reply nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py]: Done

File: nemo/collections/nlp/models/language_modeling/megatron/gpt_model.py (hunk 2 above)
Comment (shan18) [reply nemo/collections/nlp/models/language_modeling/megatron/gpt_model.py]: Done. Changed to `rotary_base`

**Review by titu1994 — APPROVED** [review]
Thanks !

**Pull request merged**